fulp_modules/features/antagonists/bloodsuckers/code/structures/bloodsucker_crypt.dm

```diff
@@ -410,65 +407,84 @@
 	return bloodsuckerdatum.AmValidAntag(target)
 
 ////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+//Subtype for bloodsucker lighting structures (candelabrum and blazier)
 
-/obj/structure/bloodsucker/candelabrum
-	name = "candelabrum"
-	desc = "It burns slowly and doesn't radiate heat."
+/obj/structure/bloodsucker/lighting
+	name = "NONDESCRIPT BLOODSUCKER LIGHTING FIXTURE THAT SHOULDN'T EXIST"
+	desc = "It burns slowly and doesn't radiate any heat"
 	icon = 'fulp_modules/features/antagonists/bloodsuckers/icons/vamp_obj.dmi'
-	icon_state = "candelabrum"
 	light_color = "#66FFFF"//LIGHT_COLOR_BLUEGREEN // lighting.dm
-	light_power = 3
-	light_range = 0 // to 2
+	light_power = 1
+	light_range = 0
 	density = FALSE
 	anchored = FALSE
-	ghost_desc = "This is a magical candle which drains the sanity of non-Bloodsuckers and non-vassals.\n\
+	ghost_desc = "This magical light drains the sanity of non-Bloodsuckers and non-vassals.\n\
 		Only bloodsuckers and their vassals can toggle it."
-	vamp_desc = "This is a magical candle which drains at the sanity of unvassalized mortals while it is active.\n\
-		Only you and your vassals can toggle it."
-	vassal_desc = "This is a magical candle which drains at the sanity of the fools who havent yet accepted your master while active.\n\
+	vamp_desc = "This magical light drains the sanity of unvassalized mortals while active.\n\
+		Only you and your vassals can toggle it.\n\
+		You alone can also toggle it from afar by <b>double-clicking</b> it."
+	vassal_desc = "This is magical light drains the sanity of those fools who havent yet accepted your master while active.\n\
 		Only you, your master, and your master's other devotees may toggle it."
-	hunter_desc = "This is a blue candelabrum, which causes insanity to those near it while active."
+	hunter_desc = "This light causes insanity to those near it while active."
 	var/lit = FALSE
+	var/active_light_range = 3
 
-/obj/structure/bloodsucker/candelabrum/Destroy()
+/obj/structure/bloodsucker/lighting/Destroy()
+	. = ..()
 	STOP_PROCESSING(SSobj, src)
-	return ..()
 
-/obj/structure/bloodsucker/candelabrum/update_icon_state()
-	icon_state = "candelabrum[lit ? "_lit" : ""]"
-	return ..()
+/obj/structure/bloodsucker/lighting/update_icon_state()
+	. = ..()
+	icon_state = "[name][lit ? "_lit" : ""]"
```

Comment:
This should use ``base_icon_state`` instead of name

Reply:
Done— I think I implemented it correctly.